### ISSUE TITLE
There are a couple of questions about accuracy

### ISSUE THREAD

**Issue by ZincJT** [isds-neu/PhyCRNet#14]
Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
        First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
        I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
        model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
        At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
        Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!

**Comment by paulpuren** [comment]
> 

Hi, I think the learning rate may be the issue for your test. Could you decrease it to 9e-4 or something like that?

**Comment by ZincJT** [comment]
Hello author, I would like to ask why the initialization parameters of the burgers equation are set this way in the def initialize_weights function. Can you tell me how you got it? I found that I replaced some of the other initialization Settings with poor results. Thank you very much for your help！！！！！！

**Comment by YHzhang2021** [comment]
Hello, author. I read your article carefully and then tried to reproduce your code. When I train it with 100 time-step arithmetic examples, the training error is basically around 1e-1, while it can easily reach 1e-5~1e-3 in the results given in your article. can you provide a little advice?
Thank you for helping me solve these problems, thank you very much!
![train loss](https://github.com/isds-neu/PhyCRNet/assets/96803975/4768dbb1-1962-474e-83f3-ff9c0cfe3795)
![x=32,y=32](https://github.com/isds-neu/PhyCRNet/assets/96803975/b4ef22f1-b0ae-4c55-86cd-0989c1c50156)

**Comment by YouqiongLiu** [comment]
> ```
>     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
>     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
>     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
>     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
>     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
>     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> ```

Hi，I was trying to run the model, but I failed. It reminded: 'c' argument has 16641 elements, which is inconsistent with 'x' and 'y' with size 16384, on Line 506. How did you change it？

**Comment by ZincJT** [comment]
> > ```
> >     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
> >     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
> >     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
> >     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
> >     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
> >     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> > ```
> 
> 嗨，我试图运行模型，但我失败了。它提醒说：“c”参数有 16641 个元素，这与第 16384 行大小为 506 的“x”和“y”不一致。你是怎么改变的？

Hello, the post_process function is used to draw pictures, I delete the drawing code can run, but for the results in the paper, I can not run out, looking forward to you can run the results, and share the operation steps.

**Comment by YouqiongLiu** [comment]
> > > ```
> > >     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
> > >     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
> > >     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
> > >     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
> > >     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
> > >     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> > > ```
> > 
> > 
> > 嗨，我试图运行模型，但我失败了。它提醒说：“c”参数有 16641 个元素，这与第 16384 行大小为 506 的“x”和“y”不一致。你是怎么改变的？
> 
> Hello, the post_process function is used to draw pictures, I delete the drawing code can run, but for the results in the paper, I can not run out, looking forward to you can run the results, and share the operation steps.


Thanks. lltry

**Comment by paulpuren** [comment]
> Hello author, I would like to ask why the initialization parameters of the burgers equation are set this way in the def initialize_weights function. Can you tell me how you got it? I found that I replaced some of the other initialization Settings with poor results. Thank you very much for your help！！！！！！

Hi, we think it might be helpful to initialize the network parameters to be close to zeros since there is only physics loss. If there is data loss, other initialization methods might also work. That is an empirical finding.

**Comment by paulpuren** [comment]
> Hello, author. I read your article carefully and then tried to reproduce your code. When I train it with 100 time-step arithmetic examples, the training error is basically around 1e-1, while it can easily reach 1e-5~1e-3 in the results given in your article. can you provide a little advice? Thank you for helping me solve these problems, thank you very much! ![train loss](https://private-user-images.githubusercontent.com/96803975/276918939-4768dbb1-1962-474e-83f3-ff9c0cfe3795.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNDUwNjIsIm5iZiI6MTcwMTE0NDc2MiwicGF0aCI6Ii85NjgwMzk3NS8yNzY5MTg5MzktNDc2OGRiYjEtMTk2Mi00NzRlLTgzZjMtZmY5YzBjZmUzNzk1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDA0MTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjOTAwYTRkMzY1YTk1MDUyZjA3OTY1NDFiODg1NGVlMjEyNjE1ZWZiMTMzZmFiNWYyZjY5ZTkxOTJkNGY2MzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9ym7TV4v1mKW0koPWes2bq8yRPpZbUgJ_rJslTY_3S8) ![x=32,y=32](https://private-user-images.githubusercontent.com/96803975/276918986-b4ef22f1-b0ae-4c55-86cd-0989c1c50156.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNDUwNjIsIm5iZiI6MTcwMTE0NDc2MiwicGF0aCI6Ii85NjgwMzk3NS8yNzY5MTg5ODYtYjRlZjIyZjEtYjBhZS00YzU1LTg2Y2QtMDk4OWMxYzUwMTU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDA0MTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmZjVjYjY3NTc2MWMwN2QzZjUwNjIyYWEyNzFlNTc4MTc1MTMwMDdlOWNmMjExYzM4OTRjMTQ2NTNhZWU1MmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M7oq_zNt4Gf9P2zwjRKFrfyKw69wZVZYZ91XnueihGk)

The training loss is still decreasing drastically. Can you train your model to 10k epochs?

**Comment by paulpuren** [comment]
> > > ```
> > >     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
> > >     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
> > >     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
> > >     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
> > >     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
> > >     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> > > ```
> > 
> > 
> > 嗨，我试图运行模型，但我失败了。它提醒说：“c”参数有 16641 个元素，这与第 16384 行大小为 506 的“x”和“y”不一致。你是怎么改变的？
> 
> Hello, the post_process function is used to draw pictures, I delete the drawing code can run, but for the results in the paper, I can not run out, looking forward to you can run the results, and share the operation steps.

Sorry for the late reply. Can you make the code work on your side now? If not, can you send more details to my email: ren.pu@northeastern.edu? Thanks.

**Comment by YouqiongLiu** [comment]
> > > ```
> > >     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
> > >     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
> > >     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
> > >     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
> > >     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
> > >     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> > > ```
> > 
> > 
> > 嗨，我试图运行模型，但我失败了。它提醒说：“c”参数有 16641 个元素，这与第 16384 行大小为 506 的“x”和“y”不一致。你是怎么改变的？
> 
> Hello, the post_process function is used to draw pictures, I delete the drawing code can run, but for the results in the paper, I can not run out, looking forward to you can run the results, and share the operation steps.

Modified the grid in the post_process function, so that the shape of x_star is [129,129], the same as the shape of u_pred.

**Comment by ZincJT** [comment]
> > ```
> >     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
> >     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
> >     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
> >     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
> >     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
> >     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> > ```
> 
> Hi ZincJT, For 2D Burgers' equations, Is the truth reference solution calculated by 'Burgers_2d_solver_[HighOrder].py' in the folder Datasets consistent with those in the paper?

It should be consistent, and it should look like the picture in the paper

**Comment by YouqiongLiu** [comment]
> > > ```
> > >     Hello author, I have read your paper carefully and also reproduced the code. Following the steps you responded to in the other question, I ran the code. 
> > >     First, I set time_batch_size=100, time_steps=101, and n_iters_adam=5000. I trained this network to get checkpoint100.pt, I tested step 100 with an accuracy of u=0.5% and v=0.8%. But when I went to get checkpoint200.pt, the accuracy was only u=7% and v=3.5%. Is there something wrong with me that leads to the low accuracy? Because I feel that when time_batch_size=1000, the accuracy will be worse and cannot reach the accuracy in the paper.
> > >     I feel that there is something wrong with my running the code step, and I deleted it when time_batch_size=100
> > >     model, optimizer, scheduler = load_checkpoint(model, optimizer, scheduler, pre_model_save_path)
> > >     At time_batch_size=200, I use checkpoint100.pt to load the model to train checkpoint200.pt. And added the code above.
> > >     Can you help me see if there is a problem with the code running steps! At the same time, I set the learning rate to 1e-3, whether the learning rate also affects the result.Thank you for helping me solve these problems, thank you very much!
> > > ```
> > 
> > 
> > Hi ZincJT, For 2D Burgers' equations, Is the truth reference solution calculated by 'Burgers_2d_solver_[HighOrder].py' in the folder Datasets consistent with those in the paper?
> 
> It should be consistent, and it should look like the picture in the paper

Hi ZincJT, 
For 2D Burgers' equations, the truth reference solution is calculated by 'Burgers_2d_solver_[HighOrder].py' in the folder Datasets. The truth reference solution (t=2) is shown as follow,
![uv_ i=1000](https://github.com/isds-neu/PhyCRNet/assets/88413314/2848fd33-1dde-41ff-9265-22b1a151549d)
Is that consistent with the results you ran?

By the following procedure, I can get reference solution u at x=32,y=32. Is that consistent with the results you ran?

 ######### download the ground truth data ############
    data_dir = '../data/2dBurgers/burgers_1501x2x128x128.mat'    
    data = scipy.io.loadmat(data_dir)
    uv = data['uv'] # [t,c,h,w]

    #(1001, 2, 128, 128)
    truth = uv[0:1001,:,:,:]

    # [1001, 2, 131, 131]
    truth = np.concatenate((truth[:, :, :, -1:], truth, truth[:, :, :, 0:2]), axis=3)
    truth = np.concatenate((truth[:, :, -1:, :], truth, truth[:, :, 0:2, :]), axis=2)
   
    u_true = truth[:, 0, :, :]

    t_true = np.linspace(0, 2, 1001)

    plt.plot(t_true, u_true[:, 32, 32], '--', label='x=32, y=32, Ref.')
    plt.xlabel('t')
    plt.ylabel('u')
    plt.xlim(0, 2)
    plt.legend()
###################################################
![Reference_xy32](https://github.com/isds-neu/PhyCRNet/assets/88413314/d3588224-0528-45c3-bc1a-5551cd438c1e)
